core/trino-main/src/main/java/io/trino/operator/output/PositionsAppenderPageBuilder.java

```diff
@@ -135,6 +135,21 @@ public boolean isEmpty()
         return declaredPositions == 0;
     }
 
+    public boolean shouldFlushBeforeRelease()
```

Comment:
why only dictionaries and not RLEs too?

nit: Generally, I've been thinking that we could have something like:
```
boolean UnnestingPositionsAppender#offer(io.trino.spi.block.Block source)
```
which would return `false` if appender would be flattened. This way we could flush RLE or dictionaries even if full page was not collected.

Reply:
> why only dictionaries and not RLEs too?

I thought about doing that but decided against it because in theory, RLE's _could_ see the same value on the next input from another driver after reuse whereas it's not possible for the same to occur for dictionaries- and we already have the "direct size limit if flattened" logic for RLE's too which somewhat mitigates the worst case of what will happen if we don't see the same RLE value when reused.

> nit: Generally, I've been thinking that we could have something like:
> 
> ```
> boolean UnnestingPositionsAppender#offer(io.trino.spi.block.Block source)
> ```
> 
> which would return `false` if appender would be flattened. This way we could flush RLE or dictionaries even if full page was not collected.

I gave that some thought, but decided against it because flushing the current page instead of flattening is something that you would have to check for each column on every input and could result in constant small page flushes due to single columns. This approach is a compromise in that we're only going to force a flush at the end of a driver processing and not on each input.